[rootx main] Clang-Tidy Clazy Warnings

```
/opt/root_src/rootx/src/rootx.cxx:148:17: warning: Potential leak of memory pointed to by 'env' [clang-analyzer-cplusplus.NewDeleteLeaks]
 1: Calling 'SetRootSys' in /opt/root_src/rootx/src/rootx.cxx:297
 2: Assuming 'exepath' is non-null in /opt/root_src/rootx/src/rootx.cxx:133
 3: Left side of '&&' is true in /opt/root_src/rootx/src/rootx.cxx:133
 4: Assuming the condition is true in /opt/root_src/rootx/src/rootx.cxx:133
 5: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:133
 6: Assuming 's' is non-null in /opt/root_src/rootx/src/rootx.cxx:138
 7: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:138
 8: Assuming 's' is non-null in /opt/root_src/rootx/src/rootx.cxx:140
 9: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:140
10: Memory is allocated in /opt/root_src/rootx/src/rootx.cxx:143
11: Potential leak of memory pointed to by 'env' in /opt/root_src/rootx/src/rootx.cxx:148
/opt/root_src/rootx/src/rootx.cxx:213:1: warning: Potential leak of memory pointed to by 'msg' [clang-analyzer-cplusplus.NewDeleteLeaks]
 1: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:299
 2: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:299
 3: Assuming 'i' is >= 'argc' in /opt/root_src/rootx/src/rootx.cxx:313
 4: Loop condition is false. Execution continues on line 328 in /opt/root_src/rootx/src/rootx.cxx:313
 5: 'notebook' is <= 0 in /opt/root_src/rootx/src/rootx.cxx:328
 6: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:328
 7: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:358
 8: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:358
 9: 'batch' is false in /opt/root_src/rootx/src/rootx.cxx:362
10: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:362
11: 'about' is false in /opt/root_src/rootx/src/rootx.cxx:364
12: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:364
13: 'batch' is false in /opt/root_src/rootx/src/rootx.cxx:369
14: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:369
15: 'about' is false in /opt/root_src/rootx/src/rootx.cxx:370
16: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:370
17: Assuming 'gNoLogo' is true in /opt/root_src/rootx/src/rootx.cxx:374
18: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:374
19: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:420
20: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:420
21: Assuming 'gChildpid' is <= 0 in /opt/root_src/rootx/src/rootx.cxx:423
22: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:423
23: Loop condition is false. Execution continues on line 457 in /opt/root_src/rootx/src/rootx.cxx:455
24: Calling 'SetLibraryPath' in /opt/root_src/rootx/src/rootx.cxx:461
25: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:190
26: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:190
27: Memory is allocated in /opt/root_src/rootx/src/rootx.cxx:195
28: Potential leak of memory pointed to by 'msg' in /opt/root_src/rootx/src/rootx.cxx:213
/opt/root_src/rootx/src/rootx.cxx:351:7: warning: Potential leak of memory pointed to by 'argvv' [clang-analyzer-cplusplus.NewDeleteLeaks]
 1: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:299
 2: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:299
 3: Assuming 'i' is < 'argc' in /opt/root_src/rootx/src/rootx.cxx:313
 4: Loop condition is true.  Entering loop body in /opt/root_src/rootx/src/rootx.cxx:313
 5: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:314
 6: Left side of '||' is false in /opt/root_src/rootx/src/rootx.cxx:314
 7: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:314
 8: Left side of '||' is false in /opt/root_src/rootx/src/rootx.cxx:314
 9: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:315
10: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:314
11: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:319
12: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:319
13: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:320
14: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:320
15: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:321
16: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:321
17: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:322
18: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:322
19: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:323
20: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:323
21: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:324
22: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:324
23: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:325
24:  Execution continues on line 328 in /opt/root_src/rootx/src/rootx.cxx:325
25: 'notebook' is > 0 in /opt/root_src/rootx/src/rootx.cxx:328
26: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:328
27: Memory is allocated in /opt/root_src/rootx/src/rootx.cxx:341
28: 'i' is < 'numnbargs' in /opt/root_src/rootx/src/rootx.cxx:343
29: Loop condition is true.  Entering loop body in /opt/root_src/rootx/src/rootx.cxx:343
30: Assuming 'i' is >= 'numnbargs' in /opt/root_src/rootx/src/rootx.cxx:343
31: Loop condition is false. Execution continues on line 345 in /opt/root_src/rootx/src/rootx.cxx:343
32: Potential leak of memory pointed to by 'argvv' in /opt/root_src/rootx/src/rootx.cxx:351
/opt/root_src/rootx/src/rootx.cxx:468:4: warning: Potential leak of memory pointed to by 'argvv' [clang-analyzer-cplusplus.NewDeleteLeaks]
 1: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:299
 2: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:299
 3: Assuming 'i' is >= 'argc' in /opt/root_src/rootx/src/rootx.cxx:313
 4: Loop condition is false. Execution continues on line 328 in /opt/root_src/rootx/src/rootx.cxx:313
 5: 'notebook' is <= 0 in /opt/root_src/rootx/src/rootx.cxx:328
 6: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:328
 7: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:358
 8: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:358
 9: 'batch' is false in /opt/root_src/rootx/src/rootx.cxx:362
10: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:362
11: 'about' is false in /opt/root_src/rootx/src/rootx.cxx:364
12: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:364
13: 'batch' is false in /opt/root_src/rootx/src/rootx.cxx:369
14: Taking true branch in /opt/root_src/rootx/src/rootx.cxx:369
15: 'about' is false in /opt/root_src/rootx/src/rootx.cxx:370
16: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:370
17: Assuming 'gNoLogo' is true in /opt/root_src/rootx/src/rootx.cxx:374
18: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:374
19: Assuming the condition is false in /opt/root_src/rootx/src/rootx.cxx:420
20: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:420
21: Assuming 'gChildpid' is <= 0 in /opt/root_src/rootx/src/rootx.cxx:423
22: Taking false branch in /opt/root_src/rootx/src/rootx.cxx:423
23: Memory is allocated in /opt/root_src/rootx/src/rootx.cxx:443
24: Loop condition is false. Execution continues on line 457 in /opt/root_src/rootx/src/rootx.cxx:455
25: Potential leak of memory pointed to by 'argvv' in /opt/root_src/rootx/src/rootx.cxx:468
/opt/root_src/rootx/src/rootxx.cxx:783:10: warning: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 [clang-analyzer-security.insecureAPI.strcpy]
 1: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 in /opt/root_src/rootx/src/rootxx.cxx:783
/opt/root_src/rootx/src/rootx.cxx
/opt/root_src/rootx/src/rootxx.cxx
```

```
/opt/root_src/core/base/inc/TVersionCheck.h:31:1: warning: non-POD static (TVersionCheck) [clazy-non-pod-global-static]
/opt/root_src/main/src/nbmain.cxx:49:1: warning: non-POD static (basic_string) [clazy-non-pod-global-static]
/opt/root_src/main/src/nbmain.cxx:224:4: warning: Potential leak of memory pointed to by 'jargv' [clang-analyzer-cplusplus.NewDeleteLeaks]
 1: Taking false branch in /opt/root_src/main/src/nbmain.cxx:195
 2: Taking false branch in /opt/root_src/main/src/nbmain.cxx:204
 3: Memory is allocated in /opt/root_src/main/src/nbmain.cxx:213
 4: Assuming 'n' is >= 'argc' in /opt/root_src/main/src/nbmain.cxx:216
 5: Loop condition is false. Execution continues on line 218 in /opt/root_src/main/src/nbmain.cxx:216
 6: Potential leak of memory pointed to by 'jargv' in /opt/root_src/main/src/nbmain.cxx:224
/opt/root_src/main/src/pmain.cxx:59:11: warning: Potential leak of memory pointed to by 'ev' [clang-analyzer-cplusplus.NewDeleteLeaks]
 1: Assuming 'argc' is >= 6 in /opt/root_src/main/src/pmain.cxx:178
 2: Taking true branch in /opt/root_src/main/src/pmain.cxx:178
 3: Calling 'ReadPutEnvs' in /opt/root_src/main/src/pmain.cxx:180
 4: Assuming 'envfile' is non-null in /opt/root_src/main/src/pmain.cxx:51
 5: Left side of '||' is false in /opt/root_src/main/src/pmain.cxx:51
 6: Assuming the condition is false in /opt/root_src/main/src/pmain.cxx:51
 7: Taking false branch in /opt/root_src/main/src/pmain.cxx:51
 8: Assuming 'fenv' is non-null in /opt/root_src/main/src/pmain.cxx:55
 9: Taking false branch in /opt/root_src/main/src/pmain.cxx:55
10: Loop condition is true.  Entering loop body in /opt/root_src/main/src/pmain.cxx:59
11: Assuming 'l' is > 0 in /opt/root_src/main/src/pmain.cxx:62
12: Left side of '&&' is true in /opt/root_src/main/src/pmain.cxx:62
13: Assuming the condition is false in /opt/root_src/main/src/pmain.cxx:62
14: Taking false branch in /opt/root_src/main/src/pmain.cxx:62
15: 'l' is > 0 in /opt/root_src/main/src/pmain.cxx:64
16: Left side of '||' is false in /opt/root_src/main/src/pmain.cxx:64
17: Assuming the condition is false in /opt/root_src/main/src/pmain.cxx:64
18: Taking false branch in /opt/root_src/main/src/pmain.cxx:64
19: Assuming the condition is false in /opt/root_src/main/src/pmain.cxx:66
20: Taking false branch in /opt/root_src/main/src/pmain.cxx:66
21: Memory is allocated in /opt/root_src/main/src/pmain.cxx:68
22: Potential leak of memory pointed to by 'ev' in /opt/root_src/main/src/pmain.cxx:59
/opt/root_src/main/src/pmain.cxx:69:7: warning: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 [clang-analyzer-security.insecureAPI.strcpy]
 1: Call to function 'strcpy' is insecure as it does not provide bounding of the memory buffer. Replace unbounded copy functions with analogous functions that support length arguments such as 'strlcpy'. CWE-119 in /opt/root_src/main/src/pmain.cxx:69
/opt/root_src/core/base/inc/TVersionCheck.h
/opt/root_src/main/src/nbmain.cxx
/opt/root_src/main/src/pmain.cxx
```

--- a/main/src/nbmain.cxx
+++ b/main/src/nbmain.cxx
@@ -43,10 +43,10 @@ using namespace std;
 
 #ifdef WIN32
 #include <process.h>
-static string pathsep("\\");
+std::string pathsep = "\\";
 #define execlp _execlp
 #else
-static string pathsep("/");
+std::string pathsep = "/";
 #endif
 
 ////////////////////////////////////////////////////////////////////////////////
@@ -224,5 +224,6 @@ int main(int argc, char **argv)
    fprintf(stderr,
            "Error starting ROOT notebook -- please check that Jupyter is installed\n");
 
+   delete [] jargv;
    return 1;
 }

--- a/main/src/nbmain.cxx
+++ b/main/src/nbmain.cxx
@@ -43,10 +43,10 @@ using namespace std;
 
 #ifdef WIN32
 #include <process.h>
-std::string pathsep = "\\";
+constexpr const char *pathsep = "\\";
 #define execlp _execlp
 #else
-std::string pathsep = "/";
+constexpr const char *pathsep = "/";
 #endif
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/main/src/pmain.cxx
+++ b/main/src/pmain.cxx
@@ -34,6 +34,7 @@
 #include "TInterpreter.h"
 #include "TROOT.h"
 #include "TSystem.h"
+#include "strlcpy.h"
 
 
 static Int_t gLogLevel = 0;
@@ -66,8 +67,8 @@ static void ReadPutEnvs(const char *envfile)
       if (!strchr(ln, '=')) continue;
       // Good line
       char *ev = new char[l+1];
-      strcpy(ev, ln);
-      putenv(ev);
+      strlcpy(ev, ln, l+1);
+      putenv(ev); // NOLINT: do not release memory in ev, will be used by environment variable
    }
 
    // Close the file

--- a/rootx/src/rootx.cxx
+++ b/rootx/src/rootx.cxx
@@ -142,7 +142,7 @@ static void SetRootSys()
             int l2 = strlen(ep) + 10;
             char *env = new char[l2];
             snprintf(env, l2, "ROOTSYS=%s", ep);
-            putenv(env);
+            putenv(env); // NOLINT: allocated memory now used by environment variable
          }
       }
       delete [] ep;
@@ -351,6 +351,8 @@ int main(int argc, char **argv)
       fprintf(stderr, "%s: can't start ROOT notebook -- this option is only available when building with CMake, please check that %s exists\n",
               argv[0], arg0);
 
+      delete [] argvv;
+
       return 1;
    }
 
@@ -467,6 +469,8 @@ int main(int argc, char **argv)
    // Exec failed
    fprintf(stderr, "%s: can't start ROOT -- check that %s exists!\n",
            argv[0], arg0);
+
+   delete [] argvv;
 
    return 1;
 }

--- a/rootx/src/rootxx.cxx
+++ b/rootx/src/rootxx.cxx
@@ -30,6 +30,7 @@
 
 #include "Rtypes.h"
 #include "snprintf.h"
+#include "strlcpy.h"
 
 #include "rootcoreteam.h"
 
@@ -779,8 +780,9 @@ int DrawCredits(bool draw, bool extended)
 
       struct passwd *pwd = getpwuid(getuid());
       if (pwd) {
-         char *name = new char [strlen(pwd->pw_gecos)+1];
-         strcpy(name, pwd->pw_gecos);
+         size_t sz = strlen(pwd->pw_gecos)+1;
+         char *name = new char [sz];
+         strlcpy(name, pwd->pw_gecos, sz);
          char *s = strchr(name, ',');
          if (s) *s = 0;
          char line[1024];

